src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java

```diff
@@ -729,8 +745,25 @@ public CQLSSTableWriter build()
                     Schema.instance.submit(SchemaTransformations.addTable(tableMetadata, true));
                 }
 
+                if (compressionDictionary != null)
+                {
+                    CompressionParams compressionParams = tableMetadata.params.compression;
+
+                    if (!compressionParams.isDictionaryCompressionEnabled())
+                    {
+                        throw new IllegalStateException("Table's compressor can not accept any dictionary: " + compressionParams.asMap());
+                    }
+
+                    IDictionaryCompressor compressor = (IDictionaryCompressor) compressionParams.getSstableCompressor();
+                    if (!compressor.canConsumeDictionary(compressionDictionary))
+                    {
+                        throw new IllegalStateException("Provided dictionary can not be consumed by table's compressor. Dictionary type: " + compressionDictionary.kind()
+                                                        + ", compressor can accept type: " + compressor.acceptableDictionaryKind());
```

Comment:
nit: proposing the revised error message 
```suggestion
                        throw new IllegalStateException("Provided dictionary can not be consumed by table's compressor. " +
                                                        "Provided dictionary type: " + compressionDictionary.kind()
                                                        + "; expected dictionary type by the compressor: " + compressor.acceptableDictionaryKind());
```